XamarinCommunityToolkit/Behaviors/TagBehavior.shared.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text.RegularExpressions;
+using System.Windows.Input;
+using Xamarin.Forms;
+
+namespace Microsoft.Toolkit.Xamarin.Forms.Behaviors
+{
+    public class TagBehavior : BaseBehavior<Label>
+    {
+        public IList<TagType> TagTypes { get; } = new List<TagType>();
+
+        string RegexPattern => $@"[{string.Join(string.Empty, TagTypes.Select(s => s.Symbol))}]\w+";
+
+        public static readonly BindableProperty CommandProperty =
+         BindableProperty.Create(nameof(Command), typeof(ICommand), typeof(TagBehavior), null, defaultBindingMode: BindingMode.TwoWay);
+
+        public ICommand Command
+        {
+            get => (ICommand)GetValue(CommandProperty);
+            set => SetValue(CommandProperty, value);
+        }
+
+        protected override void OnAttachedTo(Label bindable)
+        {
+            OnViewPropertyChanged(bindable, new PropertyChangedEventArgs(nameof(Label.FormattedText)));
+            base.OnAttachedTo(bindable);
+        }
+
+        protected override void OnViewPropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            base.OnViewPropertyChanged(sender, e);
+
+            var label = sender as Label;
+
+            if (e.PropertyName == Label.FormattedTextProperty.PropertyName && $"{label.FormattedText}" is string textValue && !string.IsNullOrEmpty(textValue))
+            {
+                label.FormattedText.Spans.Clear();
+                var formatted = label.FormattedText;
+                var collection = Regex.Matches(textValue, RegexPattern, RegexOptions.Singleline);
+
+                var lastIndex = 0;
+
+                foreach (Match item in collection)
+                {
+                    var text = textValue.Substring(lastIndex, item.Index - lastIndex);
+                    formatted.Spans.Add(CreateSpan(text));
+                    lastIndex = item.Index + item.Length;
+
+                    var span = CreateSpan(item.Value, true);
+
+                    formatted.Spans.Add(span);
+
+                    if (Command != null)
+                    {
+                        AddGestureRecognizer(span);
+                    }
+
+                }
+
+                var remainingText = textValue.Substring(lastIndex);
+
+                formatted.Spans.Add(CreateSpan(remainingText));
+            }
+        }
+
+        protected override void OnPropertyChanged([CallerMemberName] string propertyName = null)
+        {
+            base.OnPropertyChanged(propertyName);
+
+            if (propertyName == CommandProperty.PropertyName && Command != null)
+            {
```

Comment:
we are trying to not using parenthesis for one line if-else.

Reply:
Fixed